src/components/__tests__/FormInviteFriends.cy.js

```diff
@@ -0,0 +1,175 @@
+
+const { getPaletteColor } = colors;
+const black = getPaletteColor('black');
+const grey10 = getPaletteColor('grey-10');
+
+describe('<FormInviteFriends>', () => {
+  it('has translation for all strings', () => {
+    cy.testLanguageStringsInContext(
+      [
+        'buttonInviteFriends',
+        'labelInviteEmailAddresses',
+        'labelLanguage',
+        'messageRequiredAddresses',
+        'textMessage',
+        'titleMessage',
+      ],
+      'onboarding',
+      i18n,
+    );
+    cy.testLanguageStringsInContext(['messageFieldRequired'], 'form', i18n);
+  });
+
+  context('desktop', () => {
+    beforeEach(() => {
+      cy.mount(FormInviteFriends, {
+        props: {},
+      });
+      cy.viewport('macbook-16');
+    });
+
+    it('renders component', () => {
+      cy.dataCy('form-invite-friends').should('be.visible');
+      cy.dataCy('invite-description')
+        .should('be.visible')
+        .and('have.css', 'font-size', '14px')
+        .and('have.css', 'font-weight', '400')
+        .and('have.css', 'margin-bottom', '24px')
+        .and('have.color', grey10);
+      cy.dataCy('invite-email-addresses')
+        .should('be.visible')
+        .and('have.css', 'margin-top', '16px')
+        .and('have.css', 'margin-bottom', '16px')
+        .find('label[for="invite-email-addresses"]')
+        .should('be.visible')
+        .and('have.css', 'font-size', '12px')
+        .and('have.css', 'font-weight', '700')
+        .and('have.color', grey10);
+      cy.dataCy('invite-email-addresses-input').should('be.visible');
+      cy.dataCy('invite-language')
+        .should('be.visible')
+        .and('have.css', 'margin-top', '16px')
+        .and('have.css', 'margin-bottom', '16px')
+        .find('label[for="select-language"]')
+        .should('be.visible')
+        .and('have.css', 'font-size', '12px')
+        .and('have.css', 'font-weight', '700')
+        .and('have.color', grey10);
+      cy.dataCy('title-message')
+        .should('be.visible')
+        .and('have.css', 'font-size', '14px')
+        .and('have.css', 'font-weight', '700')
+        .and('have.color', black)
+        .then(($el) => {
+          const textContent = $el.text();
+          cy.stripHtmlTags(i18n.global.t('onboarding.titleMessage')).then(
+            (text) => {
+              expect(textContent).to.contain(text);
+            },
+          );
+        });
+      cy.dataCy('text-message')
+        .should('be.visible')
+        .and('have.css', 'font-size', '14px')
+        .and('have.css', 'font-weight', '400')
+        .and('have.color', black)
+        .then(($el) => {
+          const textContent = $el.text();
+          cy.stripHtmlTags(i18n.global.t('onboarding.textMessage')).then(
+            (text) => {
+              expect(textContent).to.contain(text);
+            },
+          );
+        });
+    });
+
+    it('allows to change the message language', () => {
+      cy.dataCy('invite-language-input').select('English');
```

Comment:
You can create one custom Cypress command here https://github.com/auto-mat/ride-to-work-by-bike-frontend/blob/main/test/cypress/support/commands.js for testing select box widget with different langs (command arguments).

Reply:
Updated:
* Dynamically generate the list of language options from i18n plugin
* Extract the testing logic into a custom command